azurerm/internal/services/network/firewall_resource.go

```diff
@@ -132,6 +132,33 @@ func resourceArmFirewall() *schema.Resource {
 				},
 			},
 
+			"sku": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.AZFWHub),
+								string(network.AZFWVNet),
+							}, false),
+						},
+
+						"tier": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.Standard),
+								string(network.Premium),
+							}, false),
+						},
+					},
+				},
+			},
```

Comment:
I think these will need to be computed to prevent resources that don't specify it having a diff? (try the `_basic` test to see what I mean)